### ISSUE TITLE
Prevent using portable mode in Windows installations

### ISSUE THREAD

**Issue by mohsenkhanpour** [Microsoft/vscode#57990]
- VSCode Version:

Version: 1.27.0 (user setup)
Commit: 7b9afc4196bda60b0facdf62cfc02815509b23d6
Date: 2018-09-05T05:29:44.670Z
Electron: 2.0.7
Chrome: 61.0.3163.100
Node.js: 8.9.3
V8: 6.1.534.41
Architecture: x64

- OS Version: Windows 10

Steps to Reproduce:

1. Clicked on the `Restart to apply update` button
2. Restarted and new window opened without my personal settings and extensions
3. When checked `data` folder, contents were remove from there.

Does this issue occur when all extensions are disabled?: No, but it happened after clicking on upgrade, during upgrade process no extension is running I suppose.

Is there any way to get my data back? I had tons of settings, extensions, and themes. I personally wouldn't like to add them one by one again.
Obviously as `data` folder is a portable settings folder and it shouldn't have been affected by the update.

**Comment by mjbvz** [comment]
By `data` folder, do you mean the one under `appdata`?

**Comment by mohsenkhanpour** [comment]
If it is worth mentioning I was on a system-wide setup, then migrated to the portable mode as it is instructed in the docs. I had no problems with updates then, but however recently I got some recurring notifications on how it is suggested to move to a user-wide setup. I removed the system-wide installation from Add/Remove Programs and installed the user-wide setup, then moved my data folder again into Code's folder.

**Comment by joaomoreno** [comment]
@mohsenkhanpour Check the docs:

> Portable Mode is supported on the ZIP download for Windows and Linux, as well as the regular Application download for macOS.

https://code.visualstudio.com/docs/editor/portable

I will leave this issue open in order to handle the case where a user might create a `data` folder on a Code installation, like you've done, and warn about that not working at all.

**Comment by joaomoreno** [comment]
You mention that you followed the docs, but the docs clearly state that you should've downloaded the ZIP and simply copy the user data and extensions directories over. It seems you simply created a `data` folder in the setup folder, is that correct?

**Comment by mohsenkhanpour** [comment]
@joaomoreno I did download the zip file at first, and I even created `Open with VS Code` manually in the registry. However as I stated I started to get a `move to user-wide` notification. It must have fooled me to suppose portable mode would work with user-wide setups.

One question though: Was it because I was using portable mode that I got the notification? or could be something in my `data` folder telling VS Code it was a system-wide installation?

Food for thought: As you see in the screen-shot the upgrade has understood that I am using a portable mode so it has created 9175 files there. However as after upgrade VS Code opens immediately they may have been created by that.

Thanks for your information, it was my own fault, but the word "installation" in the docs regarding portable mode, and the recurring notification after using the zip confused me on how to properly use the portable mode. 

> The data folder can be moved to other VS Code installations. This is useful for updating your portable Code version: simply move the data folder to a newer extracted version of VS Code.

I will download the .zip and uninstall my user-wide setup, however it would be great if we could use `data` folder for installations too. Auto-upgrade is a bliss.

**Comment by joaomoreno** [comment]
The ZIP will not tell you to move to the user-wide notification. I believe you might've opened the ZIP version while another, system-wide, version was already running. The second instance simply opened a new window on the first one... Very unfortunate.

> I will download the .zip and uninstall my user-wide setup, however it would be great if we could use data folder for installations too. Auto-upgrade is a bliss.

Thumbs up in #56326

**Comment by KubaSzostak** [comment]
The same here. I was migrating from system wide setup to portable setup v1.25. It was copy-pasted to `W:\Programs\VSCode`. Then after self-update to v1.27 my 'data' folder was deleted. After that I deleted `%APPDATA%\Code` and `%USERPROFILE%\.vscode\extensions` folders. But today after self-update to 1.27.2 'data' folder is gone again. 

So I went to _Control Panel_ and I found _Microsoft Visual Studio Code (User)_  there. 
![image](https://user-images.githubusercontent.com/4609437/45770697-651c8a00-bc43-11e8-985f-36e332ed5953.png)
I uninstalled it and now whole `W:\Programs\VSCode` is gone 😮

It would be nice if VSCode ask me if I allow it to update so I could make copy of 'data' folder before update.
